### PR TITLE
Implement remaining AddSub instructions

### PR DESCRIPTION
**Pull request by cristeahub**

### CODE REVIEW

File: src/cpu.rs [hunk 1]
```diff
@@ -117,15 +151,36 @@ mod tests {
...
+            destination_register: 3,
+        };
 
         let mut cpu = ARM7TDMI::new();
+        cpu.execute_instruction(add_register);
+        cpu.execute_instruction(sub_register);
         cpu.execute_instruction(add_immediate);
+        cpu.execute_instruction(sub_immediate);
 
         println!("{:?}", cpu);
```
Comment (aleksanb) [inline src/cpu.rs]: Could you add asserts for the registers as opposed to printing the values?
That way we can ensure the operations did the correct thing, as well as being able to accept the instructions in general.

File: src/cpu.rs [hunk 2]
```diff
@@ -49,13 +49,36 @@ impl ARM7TDMI {
     pub fn execute_instruction(&mut self, instruction: Instruction) {
         match instruction {
             AddSub { opcode, operand, source_register, destination_register } => {
+                use opcode::OpcodeAddSub::*;
                 match opcode {
                     AddRegister => {
-                        self.register_instruction(operand,
+                        let register_operand = operand;
+                        self.register_instruction(register_operand,
```
Comment (aleksanb) [inline src/cpu.rs]: Skip variable binding and put it straight into the function? (less lines of code for the human to read!)

Comment (cristeahub) [reply src/cpu.rs]: Sure!

File: src/cpu.rs (hunk 1 above)
Comment (cristeahub) [reply src/cpu.rs]: Yup!

**Comment by cristeahub** [comment]
ping!

**Pull request merged**